lib/ask/runtime/nuntium_channel.ex

```diff
@@ -192,6 +192,19 @@ defmodule Ask.Runtime.NuntiumChannel do
       end
 
     json_reply = reply_to_messages(reply, from, respondent_id, channel_id)
+
+    case json_reply do
+      [] ->
+        :ok
+
+      _ ->
+        ChannelBroker.force_active_respondent(
+          channel_id,
+          Repo.get(Respondent, respondent_id),
```

Comment:
Reading `ChanneBroker.activate_respondent/2` I see it only needs `respondent.id`; we could spare a query (we already request the respondent twice above) by passing `respondent_id` directly.